### PR TITLE
all: ability to skip autogenerated files

### LINKED ISSUE #247: all: ability to not scan autogenerated files
We should easily ignore generated go files(with `DO NOT EDIT` label at the top), in particular we might be also interested in ignoring `protobuf` files.

### PR DESCRIPTION
**Pull request by cristaloleg**
Fixes #247

### CODE REVIEW

File: cmd/criticize/main.go [hunk 1]
```diff
@@ -154,16 +156,25 @@ func (l *linter) CheckPackage(pkgPath string) {
...
+			l.checkFile(f)
+		}
+	}
+}
+
+func isGenerated(f *ast.File) bool {
+	if f.Comments == nil || len(f.Comments) == 0 {
+		return false
 	}
+	return strings.HasPrefix(f.Comments[0].Text(), "Code generated .* DO NOT EDIT.")
```
Comment (Quasilyte) [inline cmd/criticize/main.go]: Shouldn't this be a regexp?
Usually, `.*` is a program name.

File: cmd/criticize/main.go [hunk 2]
```diff
@@ -61,6 +62,7 @@ func parseArgv(l *linter) {
 	}
 
 	enable := flag.String("enable", "all", "comma-separated list of enabled checkers")
+	flag.BoolVar(&l.scanGenerated, "scanGenerated", false, "")
```
Comment (Quasilyte) [inline cmd/criticize/main.go]: Maybe `s/scan/check/`?

File: cmd/lintwalk/main.go [hunk 3]
```diff
@@ -29,10 +29,9 @@ func getPackagePrefix(dir string) string {
 
 // Main implements gocritic sub-command entry point.
 func Main() {
-	enable := flag.String("enable", "all",
-		`forwarded to linter "as is"`)
-	exclude := flag.String("exclude", "testdata/|vendor/|builtin/",
-		`regexp used to skip package names`)
+	enable := flag.String("enable", "all", `forwarded to linter "as is"`)
+	exclude := flag.String("exclude", "testdata/|vendor/|builtin/", `regexp used to skip package names`)
+	scanGenerated := flag.Bool("scanGenerated", false, ``)
```
Comment (Quasilyte) [inline cmd/lintwalk/main.go]: At least mention that this flag is forwarded to the linter.
We can change API to accept something like `-linterFlags=...` that accepts all flags to be forwarded like Go `-gcflags` do.

Comment (cristaloleg) [reply cmd/lintwalk/main.go]: Do you mean a comment?

Comment (Quasilyte) [reply cmd/lintwalk/main.go]: Flag documentation string is empty right now.
We can avoid flag duplication problem with single arg forwarding mechanism (as in Go tool `gcflags`), so we only need to document one flag, or have several corresponding forwarded flags with short documentation (like `forwarded to linter "as is"`).

Comment (Quasilyte) [reply cmd/lintwalk/main.go]:
```diff
- scanGenerated := flag.Bool("scanGenerated", false, ``)
+ scanGenerated := flag.Bool("scanGenerated", false, `forwarded to linter "as is"`)
```

**Comment by Quasilyte** [comment]
Ping.

**Pull request closed without merging**